enterprise/server/remote_execution/commandutil/commandutil.go

```diff
@@ -99,6 +104,42 @@ func Run(ctx context.Context, command *repb.Command, workDir string, stdin io.Re
 	}
 }
 
+// RunWithProcessGroupCleanup runs the given command, ensuring that child
+// processes are killed if the command times out.
+//
+// It is intended to be used with a command created via exec.Command(), not
+// exec.CommandContext(). Unlike exec.CommandContext.Run(), it kills the process
```

Comment:
just for my edification; what happens if a caller does run a exec.CommandContext cmd with this?

Reply:
If that happens, there's a race between the `exec` lib and our goroutine to run `syscall.Kill(pid, syscall.SIGKILL)` and `syscall.Kill(-pgid, syscall.SIGKILL)` respectively. No matter the order, the Wait() call will exit with a `signal: killed` error, but child processes may or may not be cleaned up properly.